authentication/src/main/java/io/camunda/authentication/service/BasicCamundaUserService.java

```diff
@@ -7,36 +7,46 @@
  */
 package io.camunda.authentication.service;
 
+import io.camunda.authentication.entity.AuthenticationContext;
 import io.camunda.authentication.entity.CamundaUser;
 import io.camunda.authentication.entity.CamundaUserDTO;
 import io.camunda.search.entities.RoleEntity;
+import java.util.Optional;
 import org.springframework.context.annotation.Profile;
+import org.springframework.security.core.Authentication;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Service;
 
 @Profile("auth-basic")
 @Service
 public class BasicCamundaUserService implements CamundaUserService {
+  private Optional<CamundaUser> getCurrentCamundaUser() {
+    return Optional.ofNullable(SecurityContextHolder.getContext().getAuthentication())
+        .map(Authentication::getPrincipal)
+        .map(principal -> principal instanceof final CamundaUser user ? user : null);
```

Comment:
```suggestion
        .filter(CamundaUser.class::isInstance);
```

Probably a matter of taste.